Failed to compile the sdk generated by model 4.3.22

## The problem

After updating the model to `4.3.22`, the generated sdk had compiling errors.

## Environment

* Desktop OS/version
> macOS Mojave (version 10.14.4)
* Java version to run maven
> openjdk version "11.0.2" 2019-01-15
* Go version under test
> go1.12.1 darwin/amd64
* The metamodel version speicifed in the root pom.xml
> 1.3.0
* The model version sepecified in the root pom.xml
> 4.3.22

## Description

When running `mvn package` command with the model 4.3.22, the compile phrase would exit unexpectedly with the following errors output.

```
[INFO] --- exec-maven-plugin:1.2.1:exec (compile) @ go-sdk ---
_/Users/joey/Work/ovirt/ovirt-engine-sdk-go/sdk/ovirtsdk
# _/Users/joey/Work/ovirt/ovirt-engine-sdk-go/sdk/ovirtsdk
ovirtsdk/services.go:40613:18: cannot use upgradeAction (type ClusterUpgradeAction) as type *ClusterUpgradeAction in assignment
ovirtsdk/services.go:40623:31: cannot use p.upgradeAction (type *ClusterUpgradeAction) as type ClusterUpgradeAction in argument to actionBuilder.UpgradeAction
```


**Expected behavior:** The compile phrase passed successfully.

**Actual behavior:** Failed to compile the sdk due to the errors above.

**Reproduces how often:** 100%

--- a/generator/src/main/java/org/ovirt/sdk/go/ServicesGenerator.java
+++ b/generator/src/main/java/org/ovirt/sdk/go/ServicesGenerator.java
@@ -326,7 +326,7 @@ public class ServicesGenerator implements GoGenerator {
         // Generate the parameter setter method
         buffer.addLine("func (p *%1$s) %2$s(%3$s %4$s) *%1$s{",
             requestClassName, paraMethodName, paraName, paraTypeReference.getText());
-        if (goTypes.isGoPrimitiveType(paraType)) {
+        if (goTypes.isGoPrimitiveType(paraType) || paraType instanceof EnumType) {
             buffer.addLine("p.%1$s = &%1$s", paraName);
         } else {
             buffer.addLine("p.%1$s = %1$s", paraName);
@@ -491,7 +491,7 @@ public class ServicesGenerator implements GoGenerator {
             .forEach(parameter -> {
                 String paraArgName = goNames.getParameterStyleName(parameter.getName());
                 String paraMethodName = goNames.getExportableMemberStyleName(parameter.getName());
-                if (goTypes.isGoPrimitiveType(parameter.getType())) {
+                if (goTypes.isGoPrimitiveType(parameter.getType()) || parameter.getType() instanceof EnumType) {
                     buffer.addLine("if p.%1$s != nil {", paraArgName);
                     buffer.addLine("  actionBuilder.%1$s(*p.%2$s);", paraMethodName, paraArgName);
                     buffer.addLine("}");

